### PR TITLE
feat: engagement sequence state search

### PR DESCRIPTION
**Pull request by asdfryan**
This PR also adds Get contact by ID for Apollo via private API, which was previously disabled. This is needed to be able to filter by contact Id in sequence state.

Tested locally + added new integration tests
image

### CODE REVIEW

File: openapi/v2/engagement/openapi.bundle.json [hunk 1]
```diff
@@ -1229,6 +1229,95 @@
         }
       ]
     },
+    "/sequence_states/_search": {
```
Comment (tomkit) [inline openapi/v2/engagement/openapi.bundle.json:1232]: If we think we'll be doing cached vs uncached, let's use the same flag we have for list endpoints

File: openapi/v2/engagement/paths/sequence_states@search.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    | Salesloft | contact_id, sequence_id |
+    | Outreach  | contact_id, sequence_id |
+
+  tags:
+    - Sequence States
+  security:
+    - x-api-key: []
+  parameters:
+    - $ref: ../../../common/components/parameters/query/include_raw_data.yaml
+    - $ref: ../../../common/components/parameters/query/page_size.yaml
```
Comment (tomkit) [inline openapi/v2/engagement/paths/sequence_states@search.yaml:20]: We should use `remote_provider_page_size` for remote and `page_size` for cached

Comment (tomkit) [reply openapi/v2/engagement/paths/sequence_states@search.yaml:20]: I.e. we shouldn't allow users to specify 1000 page size against remotes

**Review by tomkit — APPROVED** [review]

File: openapi/v2/engagement/openapi.bundle.json (hunk 1 above)
Comment (asdfryan) [reply openapi/v2/engagement/openapi.bundle.json:1235]: Done

**Pull request merged**